src/CMakeLists.txt

```diff
@@ -17,6 +17,8 @@ add_executable(projectMSDL WIN32
         RenderLoop.h
         SDLRenderingWindow.h
         SDLRenderingWindow.cpp
+        TextureSharing.h
+        TextureSharing.mm
```

Comment:
Did you check that this still compiles on Windows and Linux, as those OSes don't have Obj-C++ support?

Because on my Linux machine, I get an error configuring the project:

```
Cannot get compiler information:
	Compiler exited with error code 1: /usr/bin/c++ -xobjective-c++ -DAUDIO_IMPL_HEADER=\"AudioCaptureImpl_SDL.h\" -DPOCO_ENABLE_CPP11 -DPOCO_ENABLE_CPP14 -DPOCO_HAVE_FD_EPOLL -DPOCO_OS_FAMILY_UNIX -DPOCO_STATIC -DPROJECTMSDL_VERSION=\"2.0.0\" -DPROJECTM_STATIC_DEFINE -DXML_DTD -D_DEBUG -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -D_REENTRANT -D_THREAD_SAFE -D_XOPEN_SOURCE=500 -isystem/home/kai/Dev/install/projectM-latest-shared/include -isystem/home/kai/Dev/install/Poco/static/include -isystem/home/kai/Dev/install/libSDL/64/include -isystem/home/kai/Dev/install/libSDL/64/include/SDL2 -g -std=gnu++14 -fPIE -fopenmp -fpch-preprocess -v -dD -E
	Using built-in specs.
	COLLECT_GCC=/usr/bin/c++
	Target: x86_64-pc-linux-gnu
	Configured with: /var/tmp/portage/sys-devel/gcc-12.3.0/work/gcc-12.3.0/configure --host=x86_64-pc-linux-gnu --build=x86_64-pc-linux-gnu --prefix=/usr --bindir=/usr/x86_64-pc-linux-gnu/gcc-bin/12 --includedir=/usr/lib/gcc/x86_64-pc-linux-gnu/12/include --datadir=/usr/share/gcc-data/x86_64-pc-linux-gnu/12 --mandir=/usr/share/gcc-data/x86_64-pc-linux-gnu/12/man --infodir=/usr/share/gcc-data/x86_64-pc-linux-gnu/12/info --with-gxx-include-dir=/usr/lib/gcc/x86_64-pc-linux-gnu/12/include/g++-v12 --with-python-dir=/share/gcc-data/x86_64-pc-linux-gnu/12/python --enable-languages=c,c++,fortran --enable-obsolete --enable-secureplt --disable-werror --with-system-zlib --enable-nls --without-included-gettext --disable-libunwind-exceptions --enable-checking=release --with-bugurl=https://bugs.gentoo.org/ --with-pkgversion='Gentoo 12.3.0 p2' --with-gcc-major-version-only --disable-esp --enable-libstdcxx-time --enable-lto --disable-libstdcxx-pch --enable-shared --enable-threads=posix --enable-__cxa_atexit --enable-clocale=gnu --enable-multilib --with-multilib-list=m32,m64 --disable-fixed-point --enable-targets=all --enable-libgomp --disable-libssp --disable-libada --disable-cet --disable-systemtap --disable-valgrind-annotations --disable-vtable-verify --disable-libvtv --without-zstd --without-isl --enable-default-pie --enable-default-ssp
	Thread model: posix
	Supported LTO compression algorithms: zlib
	gcc version 12.3.0 (Gentoo 12.3.0 p2) 
	c++: error: language objective-c++ not recognized
	c++: error: language objective-c++ not recognized
```